auth/src/main/java/org/aerogear/mobile/security/AsyncSecurityCheckExecutor.java

```diff
@@ -91,26 +90,39 @@ public AsyncSecurityCheckExecutor build() {
      * Executes the checks asynchronously.
      *
      * Returns a {@link Map} containing the results of each executed test (a {@link Future}).
-     * The key of the map will be the output of {@link SecurityCheck#getName()}, while the value will be
+     * The key of the map will be the output of {@link SecurityCheck#getType()}, while the value will be
      * a {@link Map} of {@link Future} with the {@link SecurityCheckResult} of the check.
      *
      * @return {@link Map}
      */
     public Map<String, Future<SecurityCheckResult>> execute() {
-
-        final MetricsService metricsService = getMetricsService();
         final Map<String, Future<SecurityCheckResult>> res = new HashMap<>();
 
-        for (final SecurityCheck check : getChecks()) {
-            res.put(check.getName(), (executorService.submit(() -> {
-                final SecurityCheckResult result =  check.test(getContext());
-                if (metricsService != null) {
-                    metricsService.publish(new SecurityCheckResultMetric(result));
-                }
+        final Collection<SecurityCheck> checks = getChecks();
+        final List<SecurityCheckResult> results = new ArrayList<>(checks.size());
+
+        for (final SecurityCheck check : checks) {
+            final Future<SecurityCheckResult> future = executorService.submit(() -> {
+                SecurityCheckResult result = check.test(getContext());
+                tallyResultsAndPublish(result, results);
                 return result;
-            })));
+            });
+            res.put(check.getType(), future);
         }
 
         return res;
     }
+
+    private synchronized void tallyResultsAndPublish(SecurityCheckResult result, List<SecurityCheckResult> results) {
```

Comment:
@paolobueno This way it would work, however synchronising the whole method, you synchronize parts of the code that doesn't needs to be synchronized (thus slowing the execution).

I have created a PR to propose a different implementation for the batching. Would you mind giving that a look?
https://github.com/aerogear/aerogear-android-sdk/pull/129